als-actions/shared/src/main/scala/org/mulesoft/als/actions/hover/PatchedHover.scala

```diff
@@ -16,27 +16,38 @@ import org.mulesoft.amfintegration.vocabularies.integration.{AlsVocabularyRegist
 import org.mulesoft.amfintegration.vocabularies.propertyterms.patched.raml.raml10.Raml10UsesKeyTerm
 import org.yaml.model.YMapEntry
 
-class PatchedHover(provider: VocabularyProvider) {
+class PatchedHover(provider: VocabularyProvider,
+                   bu: BaseUnit,
+                   entry: YMapEntry,
```

Comment:
the reason you are forced to pass so many arguments is because you are making this class context-dependant, this class was meant to be created once and ask multiple hovers with the `getHover` method, but you are forzing one single entry to it

Reply:
position based parameters (for example `entry` and `obj`) should not be on constructor, just in the getter method
`bu` and `dialectId` should also not be in here, it should be passed to the companion object and from there to `init` so it is used to initialize the patches
and it should not be just a `bu, dialectId`, it should be a `Seq[(BaseUnit, DefinedBy)] in order to be more future proof

Reply:
done